computedfields/models.py

```diff
@@ -196,10 +207,15 @@ def update_dependent(mcs, instance, model=None, update_fields=None):
             else:
                 model = type(instance)
         updates = mcs._querysets_for_update(model, instance, update_fields).values()
-        if not updates:
-            return
-        with transaction.atomic():
-            for qs, fields in updates:
+        if updates:
+            with transaction.atomic():
+                for qs, fields in updates:
+                    for el in qs.distinct():
+                        el.save(update_fields=fields)
+        if dirty:
+            for model, data in dirty.items():
+                pks, fields = data
+                qs = model.objects.filter(pk__in=pks)
                 for el in qs.distinct():
                     el.save(update_fields=fields)
```

Comment:
Shouldn't this be wrapped in a transaction too (just as above) ?

Reply:
Well, it does not do anything beside creating DB load, if the transaction before failed (the dirty updates will not change anything, kinda a NOOP). But sure, since there is already an transaction, I can pull the code under the same umbrella.

Btw there are more places with missing transactions in `handlers.py`, but I can not add them easily there, since the actions are spread over `pre` + `do_update(with transaction)` + `post(no transaction)` signals. Maybe you have a good idea, how to solve that.